File: frontend/src/pages/FestivalGreetings.js
/**
 * Festival Greetings - Simple brand recall system
 * 
 * ONLY:
 * - Republic Day (January 26)
 * - Independence Day (August 15)
 * 
 * Features:
 * - ON/OFF toggle (Admin only)
 * - Recipient management (customers only, NO leads)
 * - Manual send trigger
 * - Simple audit log
 * 
 * NO marketing, NO tracking, NO analytics, NO customization
 */
import React, { useState, useEffect } from 'react';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { Input } from '@/components/ui/input';
import { Badge } from '@/components/ui/badge';
import { Dialog, DialogContent, DialogHeader, DialogTitle } from '@/components/ui/dialog';
import { Switch } from '@/components/ui/switch';
import {
  Flag, Users, Calendar, Send, UserPlus, Trash2, Download,
  Check, X, AlertCircle, Clock
} from 'lucide-react';
import { toast } from 'sonner';
import { useAuth } from '../contexts/AuthContext';
import axios from 'axios';

const API_URL = process.env.REACT_APP_BACKEND_URL;

// Fixed festivals - NO customization
const FESTIVALS = [
  {
    id: 'republic_day',
    name: 'Republic Day',
    date: 'January 26',
    emoji: '🇮🇳'
  },
  {
    id: 'independence_day',
    name: 'Independence Day',
    date: 'August 15',
    emoji: '🇮🇳'
  }
];

const FestivalGreetings = () => {
  const { user } = useAuth();
  
  // State
  const [config, setConfig] = useState(null);
  const [recipients, setRecipients] = useState([]);
  const [logs, setLogs] = useState([]);
  const [loading, setLoading] = useState(true);
  const [showAddRecipient, setShowAddRecipient] = useState(false);
  const [showSendConfirm, setShowSendConfirm] = useState(null);
  const [sending, setSending] = useState(false);
  
  // New recipient form
  const [newRecipient, setNewRecipient] = useState({
    name: '',
    mobile: '',
    source_type: 'customer'
  });
  
  // Company name for config
  const [companyName, setCompanyName] = useState('');
  
  const isAdmin = user?.role === 'super_admin' || user?.role === 'tenant_admin';

  useEffect(() => {
    loadConfig();
    loadRecipients();
    loadLogs();
  }, []);

  const loadConfig = async () => {
    try {
      const response = await axios.get(`${API_URL}/api/festival-greetings/config`, {
        headers: { Authorization: `Bearer ${localStorage.getItem('token')}` }
      });
      if (response.data.success) {
        setConfig(response.data.config);
        setCompanyName(response.data.config?.company_name || '');
      }
    } catch (error) {
      console.error('Error loading config:', error);
    } finally {
      setLoading(false);
    }
  };

  const loadRecipients = async () => {
    try {
      const response = await axios.get(`${API_URL}/api/festival-greetings/recipients`, {
        headers: { Authorization: `Bearer ${localStorage.getItem('token')}` }
      });
      if (response.data.success) {
        setRecipients(response.data.recipients);
      }
    } catch (error) {
      console.error('Error loading recipients:', error);
    }
  };

  const loadLogs = async () => {
    try {
      const response = await axios.get(`${API_URL}/api/festival-greetings/logs?limit=50`, {
        headers: { Authorization: `Bearer ${localStorage.getItem('token')}` }
      });
      if (response.data.success) {
        setLogs(response.data.logs);
      }
    } catch (error) {
      console.error('Error loading logs:', error);
    }
  };

  const handleToggleGreetings = async () => {
    if (!companyName.trim()) {
      toast.error('Please enter company name first');
      return;
    }
    
    const newEnabled = !(config?.is_enabled);
    
    try {
      const response = await axios.post(
        `${API_URL}/api/festival-greetings/config?is_enabled=${newEnabled}&company_name=${encodeURIComponent(companyName)}`,
        {},
        { headers: { Authorization: `Bearer ${localStorage.getItem('token')}` } }
      );
      
      if (response.data.success) {
        setConfig(response.data.config);
        toast.success(`Greetings ${newEnabled ? 'enabled' : 'disabled'}`);
      }
    } catch (error) {
      toast.error(error.response?.data?.detail || 'Failed to update config');
    }
  };

  const handleSaveCompanyName = async () => {
    if (!companyName.trim()) {
      toast.error('Company name is required');
      return;
    }
    
    try {
      const response = await axios.post(
        `${API_URL}/api/festival-greetings/config?is_enabled=${config?.is_enabled || false}&company_name=${encodeURIComponent(companyName)}`,
        {},
        { headers: { Authorization: `Bearer ${localStorage.getItem('token')}` } }
      );
      
      if (response.data.success) {
        setConfig(response.data.config);
        toast.success('Company name saved');
      }
    } catch (error) {
      toast.error('Failed to save');
    }
  };

  const handleAddRecipient = async () => {
    if (!newRecipient.name || !newRecipient.mobile) {
      toast.error('Name and mobile are required');
      return;
    }
    
    try {
      const response = await axios.post(
        `${API_URL}/api/festival-greetings/recipients`,
        newRecipient,
        { headers: { Authorization: `Bearer ${localStorage.getItem('token')}` } }
      );
      
      if (response.data.success) {
        toast.success('Recipient added');
        setShowAddRecipient(false);
        setNewRecipient({ name: '', mobile: '', source_type: 'customer' });
        loadRecipients();
      }
    } catch (error) {
      toast.error(error.response?.data?.detail || 'Failed to add recipient');
    }
  };

  const handleRemoveRecipient = async (recipientId) => {
    if (!window.confirm('Remove this recipient?')) return;
    
    try {
      await axios.delete(
        `${API_URL}/api/festival-greetings/recipients/${recipientId}`,
        { headers: { Authorization: `Bearer ${localStorage.getItem('token')}` } }
      );
      toast.success('Recipient removed');
      loadRecipients();
    } catch (error) {
      toast.error('Failed to remove');
    }
  };

  const handleImportCustomers = async () => {
    if (!window.confirm('Import all customers as greeting recipients?')) return;
    
    try {
      const response = await axios.post(
        `${API_URL}/api/festival-greetings/recipients/import-customers`,
        {},
        { headers: { Authorization: `Bearer ${localStorage.getItem('token')}` } }
      );
      
      if (response.data.success) {
        toast.success(response.data.message);
        loadRecipients();
      }
    } catch (error) {
      toast.error(error.response?.data?.detail || 'Import failed');
    }
  };

  const handleSendGreetings = async (festival) => {
    setSending(true);
    
    try {
      const response = await axios.post(
        `${API_URL}/api/festival-greetings/send?festival=${festival}`,
        {},
        { headers: { Authorization: `Bearer ${localStorage.getItem('token')}` } }
      );
      
      if (response.data.success) {
        toast.success(response.data.message);
        loadLogs();
      }
    } catch (error) {
      toast.error(error.response?.data?.detail || 'Send failed');
    } finally {
      setSending(false);
      setShowSendConfirm(null);
    }
  };

  // Get next greeting date
  const getNextGreeting = () => {
    const today = new Date();
    const year = today.getFullYear();
    
    const republicDay = new Date(year, 0, 26); // Jan 26
    const independenceDay = new Date(year, 7, 15); // Aug 15
    
    if (today < republicDay) {
      return { festival: FESTIVALS[0], date: republicDay };
    } else if (today < independenceDay) {
      return { festival: FESTIVALS[1], date: independenceDay };
    } else {
      return { festival: FESTIVALS[0], date: new Date(year + 1, 0, 26) };
    }
  };

  const nextGreeting = getNextGreeting();
  const daysUntilNext = Math.ceil((nextGreeting.date - new Date()) / (1000 * 60 * 60 * 24));

  if (loading) {
    return (
      <div className="min-h-screen flex items-center justify-center">
        <div className="text-center">
          <Flag className="w-16 h-16 text-orange-500 mx-auto mb-4 animate-pulse" />
          <p className="text-slate-500">Loading...</p>
        </div>
      </div>
    );
  }

  return (
    <div className="min-h-screen bg-gradient-to-br from-orange-50 to-green-50 p-6">
      {/* Header */}
      <div className="mb-6">
        <div className="flex items-center gap-3">
          <div className="p-3 bg-gradient-to-br from-orange-100 to-green-100 rounded-xl">
            <Flag className="w-8 h-8 text-orange-600" />
          </div>
          <div>
            <h1 className="text-2xl font-bold text-slate-800">Festival Greetings</h1>
            <p className="text-slate-500">
              Republic Day (Jan 26) & Independence Day (Aug 15) only
            </p>
          </div>
        </div>
      </div>

      <div className="grid grid-cols-1 lg:grid-cols-3 gap-6">
        {/* Left Column - Configuration */}
        <div className="space-y-6">
          {/* Status Card */}
          <Card>
            <CardHeader>
              <CardTitle className="text-lg flex items-center gap-2">
                <Calendar className="w-5 h-5 text-orange-600" />
                Greeting Status
              </CardTitle>
            </CardHeader>
            <CardContent className="space-y-4">
              {/* Next Greeting */}
              <div className="p-4 bg-gradient-to-br from-orange-50 to-green-50 rounded-xl border border-orange-200">
                <p className="text-sm text-slate-600 mb-1">Next Greeting</p>
                <p className="text-xl font-bold text-slate-800">
                  {nextGreeting.festival.emoji} {nextGreeting.festival.name}
                </p>
                <p className="text-sm text-slate-600">
                  {nextGreeting.festival.date} ({daysUntilNext} days away)
                </p>
              </div>
              
              {/* Enable/Disable Toggle */}
              {isAdmin && (
                <div className="flex items-center justify-between p-4 bg-slate-50 rounded-lg">
                  <div>
                    <p className="font-medium text-slate-800">Greetings</p>
                    <p className="text-sm text-slate-500">
                      {config?.is_enabled ? 'Enabled' : 'Disabled'}
                    </p>
                  </div>
                  <Switch
                    checked={config?.is_enabled || false}
                    onCheckedChange={handleToggleGreetings}
                    disabled={!companyName.trim()}
                  />
                </div>
              )}
              
              {/* Company Name */}
              {isAdmin && (
                <div>
                  <label className="text-sm font-medium text-slate-700 mb-1 block">
                    Company Name (for message)
                  </label>
                  <div className="flex gap-2">
                    <Input
                      placeholder="Your Company Name"
                      value={companyName}
                      onChange={(e) => setCompanyName(e.target.value)}
                    />
                    <Button onClick={handleSaveCompanyName} variant="outline">
                      Save
                    </Button>
                  </div>
                </div>
              )}
              
              {/* Message Preview */}
              <div className="p-4 bg-white border rounded-lg">
                <p className="text-xs text-slate-500 mb-2">Message Preview</p>
                <p className="text-slate-700">
                  Warm wishes on {nextGreeting.festival.name} 🇮🇳
                  <br />
                  – {companyName || 'Your Company'}
                </p>
              </div>
              
              {/* Recipients Count */}
              <div className="flex items-center justify-between p-3 bg-green-50 rounded-lg border border-green-200">
                <div className="flex items-center gap-2">
                  <Users className="w-5 h-5 text-green-600" />
                  <span className="font-medium text-green-800">Recipients</span>
                </div>
                <span className="text-2xl font-bold text-green-700">{recipients.length}</span>
              </div>
            </CardContent>
          </Card>

          {/* Festivals Info */}
          <Card>
            <CardHeader>
              <CardTitle className="text-lg">Greeting Days</CardTitle>
            </CardHeader>
            <CardContent className="space-y-3">
              {FESTIVALS.map(festival => (
                <div 
                  key={festival.id}
                  className="flex items-center justify-between p-3 bg-slate-50 rounded-lg"
                >
                  <div className="flex items-center gap-3">
                    <span className="text-2xl">{festival.emoji}</span>
                    <div>
                      <p className="font-medium text-slate-800">{festival.name}</p>
                      <p className="text-sm text-slate-500">{festival.date}</p>
                    </div>
                  </div>
                  {isAdmin && config?.is_enabled && (
                    <Button
                      size="sm"
                      variant="outline"
                      onClick={() => setShowSendConfirm(festival.id)}
                      className="text-orange-600 border-orange-300 hover:bg-orange-50"
                    >
                      <Send className="w-4 h-4 mr-1" />
                      Send
                    </Button>
                  )}
                </div>
              ))}
              <p className="text-xs text-slate-400 text-center mt-4">
                Only these two dates. No customization.
              </p>
            </CardContent>
          </Card>
        </div>

        {/* Middle Column - Recipients */}
        <Card>
          <CardHeader>
            <div className="flex items-center justify-between">
              <CardTitle className="text-lg flex items-center gap-2">
                <Users className="w-5 h-5 text-green-600" />
                Recipients ({recipients.length})
              </CardTitle>
              <div className="flex gap-2">
                {isAdmin && (
                  <>
                    <Button
                      size="sm"
                      variant="outline"
                      onClick={handleImportCustomers}
                      title="Import existing customers"
                    >
                      <Download className="w-4 h-4 mr-1" />
                      Import
                    </Button>
                    <Button
                      size="sm"
                      onClick={() => setShowAddRecipient(true)}
                      className="bg-green-600 hover:bg-green-700"
                    >
                      <UserPlus className="w-4 h-4 mr-1" />
                      Add
                    </Button>
                  </>
                )}
              </div>
            </div>
          </CardHeader>
          <CardContent>
            <div className="space-y-2 max-h-[500px] overflow-y-auto">
              {recipients.length === 0 ? (
                <div className="text-center py-8">
                  <Users className="w-12 h-12 text-slate-300 mx-auto mb-2" />
                  <p className="text-slate-500">No recipients added</p>
                  <p className="text-sm text-slate-400">
                    Add customers to receive greetings
                  </p>
                </div>
              ) : (
                recipients.map(recipient => (
                  <div 
                    key={recipient.id}
                    className="flex items-center justify-between p-3 bg-slate-50 rounded-lg hover:bg-slate-100"
                  >
                    <div>
                      <p className="font-medium text-slate-800">{recipient.name}</p>
                      <p className="text-sm text-slate-500">{recipient.mobile}</p>
                    </div>
                    <div className="flex items-center gap-2">
                      <Badge variant="outline" className="text-xs">
                        {recipient.source_type}
                      </Badge>
                      {recipient.opted_out && (
                        <Badge variant="destructive" className="text-xs">
                          Opted Out
                        </Badge>
                      )}
                      {isAdmin && (
                        <Button
                          size="sm"
                          variant="ghost"
                          className="text-red-500 hover:text-red-700"
                          onClick={() => handleRemoveRecipient(recipient.id)}
                        >
                          <Trash2 className="w-4 h-4" />
                        </Button>
                      )}
                    </div>
                  </div>
                ))
              )}
            </div>
            
            <div className="mt-4 p-3 bg-yellow-50 border border-yellow-200 rounded-lg">
              <p className="text-xs text-yellow-800">
                <AlertCircle className="w-4 h-4 inline mr-1" />
                Only customers and past customers. NO leads, NO cold numbers.
              </p>
            </div>
          </CardContent>
        </Card>

        {/* Right Column - Logs */}
        <Card>
          <CardHeader>
            <CardTitle className="text-lg flex items-center gap-2">
              <Clock className="w-5 h-5 text-slate-600" />
              Send History
            </CardTitle>
          </CardHeader>
          <CardContent>
            <div className="space-y-2 max-h-[500px] overflow-y-auto">
              {logs.length === 0 ? (
                <div className="text-center py-8">
                  <Clock className="w-12 h-12 text-slate-300 mx-auto mb-2" />
                  <p className="text-slate-500">No greetings sent yet</p>
                </div>
              ) : (
                logs.map(log => (
                  <div 
                    key={log.id}
                    className="p-3 bg-slate-50 rounded-lg border-l-4"
                    style={{ 
                      borderLeftColor: log.status === 'sent' ? '#22c55e' : '#ef4444' 
                    }}
                  >
                    <div className="flex items-center justify-between mb-1">
                      <p className="font-medium text-sm text-slate-800">
                        {log.recipient_name}
                      </p>
                      {log.status === 'sent' ? (
                        <Check className="w-4 h-4 text-green-600" />
                      ) : (
                        <X className="w-4 h-4 text-red-600" />
                      )}
                    </div>
                    <p className="text-xs text-slate-500">{log.recipient_mobile}</p>
                    <p className="text-xs text-slate-400 mt-1">
                      {log.festival === 'republic_day' ? '🇮🇳 Republic Day' : '🇮🇳 Independence Day'}
                      {' • '}
                      {new Date(log.sent_at).toLocaleDateString()}
                    </p>
                    {log.error_message && (
                      <p className="text-xs text-red-500 mt-1">{log.error_message}</p>
                    )}
                  </div>
                ))
              )}
            </div>
          </CardContent>
        </Card>
      </div>

      {/* Add Recipient Modal */}
      <Dialog open={showAddRecipient} onOpenChange={setShowAddRecipient}>
        <DialogContent className="max-w-md">
          <DialogHeader>
            <DialogTitle className="flex items-center gap-2">
              <UserPlus className="w-5 h-5 text-green-600" />
              Add Recipient
            </DialogTitle>
          </DialogHeader>
          
          <div className="space-y-4">
            <div>
              <label className="text-sm font-medium text-slate-700 mb-1 block">Name</label>
              <Input
                placeholder="Recipient name"
                value={newRecipient.name}
                onChange={(e) => setNewRecipient(prev => ({ ...prev, name: e.target.value }))}
              />
            </div>
            <div>
              <label className="text-sm font-medium text-slate-700 mb-1 block">Mobile</label>
              <Input
                placeholder="Mobile number"
                value={newRecipient.mobile}
                onChange={(e) => setNewRecipient(prev => ({ ...prev, mobile: e.target.value }))}
              />
            </div>
            <div>
              <label className="text-sm font-medium text-slate-700 mb-1 block">Source</label>
              <select
                className="w-full border rounded-md p-2 text-sm"
                value={newRecipient.source_type}
                onChange={(e) => setNewRecipient(prev => ({ ...prev, source_type: e.target.value }))}
              >
                <option value="customer">Customer</option>
                <option value="past_customer">Past Customer</option>
                <option value="internal_contact">Internal Contact</option>
              </select>
              <p className="text-xs text-red-500 mt-1">
                Leads are NOT allowed. Only customers.
              </p>
            </div>
            
            <div className="flex justify-end gap-2 pt-4">
              <Button variant="outline" onClick={() => setShowAddRecipient(false)}>
                Cancel
              </Button>
              <Button
                onClick={handleAddRecipient}
                disabled={!newRecipient.name || !newRecipient.mobile}
                className="bg-green-600 hover:bg-green-700"
              >
                Add Recipient
              </Button>
            </div>
          </div>
        </DialogContent>
      </Dialog>

      {/* Send Confirmation Modal */}
      <Dialog open={!!showSendConfirm} onOpenChange={() => setShowSendConfirm(null)}>
        <DialogContent className="max-w-md">
          <DialogHeader>
            <DialogTitle className="flex items-center gap-2">
              <Send className="w-5 h-5 text-orange-600" />
              Send Greetings
            </DialogTitle>
          </DialogHeader>
          
          <div className="space-y-4">
            <div className="p-4 bg-orange-50 rounded-lg border border-orange-200">
              <p className="font-medium text-orange-800">
                {showSendConfirm === 'republic_day' ? '🇮🇳 Republic Day' : '🇮🇳 Independence Day'}
              </p>
              <p className="text-sm text-orange-700 mt-2">
                This will send greetings to {recipients.filter(r => !r.opted_out).length} recipients.
              </p>
            </div>
            
            <div className="p-3 bg-slate-50 rounded-lg">
              <p className="text-sm text-slate-600">Message:</p>
              <p className="mt-1 text-slate-800">
                Warm wishes on {showSendConfirm === 'republic_day' ? 'Republic Day' : 'Independence Day'} 🇮🇳
                <br />
                – {companyName}
              </p>
            </div>
            
            <div className="flex justify-end gap-2 pt-4">
              <Button variant="outline" onClick={() => setShowSendConfirm(null)}>
                Cancel
              </Button>
              <Button
                onClick={() => handleSendGreetings(showSendConfirm)}
                disabled={sending}
                className="bg-orange-600 hover:bg-orange-700"
              >
                {sending ? 'Sending...' : 'Send Now'}
              </Button>
            </div>
          </div>
        </DialogContent>
      </Dialog>
    </div>
  );
};

export default FestivalGreetings;

File: backend/routes/festival_greetings.py
"""
Festival Greetings Routes
- Simple ON/OFF configuration
- Recipient management (customers only, NO leads)
- Manual trigger for testing
- ONLY Republic Day (Jan 26) and Independence Day (Aug 15)
"""
from fastapi import APIRouter, HTTPException, Request
from models.festival_greeting import (
    FestivalGreetingConfig, GreetingRecipient, GreetingRecipientCreate, GreetingLog
)
from middleware.auth import get_current_user
from datetime import datetime, timezone
from typing import Optional
import uuid

router = APIRouter(prefix="/festival-greetings", tags=["festival-greetings"])


def get_db(request: Request):
    return request.app.state.db


# Fixed message templates - NO customization
GREETING_MESSAGES = {
    "republic_day": "Warm wishes on Republic Day 🇮🇳\n– {company_name}",
    "independence_day": "Warm wishes on Independence Day 🇮🇳\n– {company_name}"
}

# Fixed dates - ONLY these two
GREETING_DATES = {
    "republic_day": "01-26",  # January 26
    "independence_day": "08-15"  # August 15
}


# ==================== CONFIGURATION ====================

@router.get("/config")
async def get_greeting_config(request: Request):
    """Get greeting configuration for tenant"""
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    config = await db.festival_greeting_configs.find_one(
        {"tenant_id": user["tenant_id"]},
        {"_id": 0}
    )
    
    if not config:
        # Return default config
        return {
            "success": True,
            "config": {
                "is_enabled": False,
                "company_name": None,
                "greeting_dates": GREETING_DATES,
                "message_preview": {
                    "republic_day": GREETING_MESSAGES["republic_day"].format(company_name="Your Company"),
                    "independence_day": GREETING_MESSAGES["independence_day"].format(company_name="Your Company")
                }
            }
        }
    
    return {
        "success": True,
        "config": config,
        "greeting_dates": GREETING_DATES,
        "message_preview": {
            "republic_day": GREETING_MESSAGES["republic_day"].format(company_name=config.get("company_name", "Your Company")),
            "independence_day": GREETING_MESSAGES["independence_day"].format(company_name=config.get("company_name", "Your Company"))
        }
    }


@router.post("/config")
async def update_greeting_config(
    request: Request,
    is_enabled: bool,
    company_name: str
):
    """
    Update greeting configuration (Admin only)
    - Just ON/OFF toggle
    - Company name for message
    - NO other customization
    """
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    if user.get("role") not in ["super_admin", "tenant_admin"]:
        raise HTTPException(status_code=403, detail="Only Admin can configure greetings")
    
    if not company_name or not company_name.strip():
        raise HTTPException(status_code=400, detail="Company name is required")
    
    config = FestivalGreetingConfig(
        tenant_id=user["tenant_id"],
        is_enabled=is_enabled,
        company_name=company_name.strip(),
        updated_by=user["user_id"]
    )
    
    # Upsert config
    await db.festival_greeting_configs.update_one(
        {"tenant_id": user["tenant_id"]},
        {"$set": config.model_dump()},
        upsert=True
    )
    
    return {
        "success": True,
        "config": config.model_dump(),
        "message": f"Greetings {'enabled' if is_enabled else 'disabled'}"
    }


# ==================== RECIPIENTS ====================

@router.get("/recipients")
async def get_recipients(
    request: Request,
    skip: int = 0,
    limit: int = 100
):
    """Get all greeting recipients"""
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    recipients = await db.greeting_recipients.find(
        {"tenant_id": user["tenant_id"], "is_active": True},
        {"_id": 0}
    ).sort("name", 1).skip(skip).limit(limit).to_list(limit)
    
    total = await db.greeting_recipients.count_documents(
        {"tenant_id": user["tenant_id"], "is_active": True}
    )
    
    return {
        "success": True,
        "recipients": recipients,
        "total": total
    }


@router.post("/recipients")
async def add_recipient(
    recipient_data: GreetingRecipientCreate,
    request: Request
):
    """
    Add a greeting recipient
    - ONLY customers, past customers, or internal contacts
    - NO leads, NO cold numbers
    """
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    # Validate source type - NO leads
    if recipient_data.source_type not in ["customer", "past_customer", "internal_contact"]:
        raise HTTPException(
            status_code=400, 
            detail="Invalid source type. Only customer, past_customer, or internal_contact allowed. NO leads."
        )
    
    # Check for duplicate mobile
    existing = await db.greeting_recipients.find_one({
        "tenant_id": user["tenant_id"],
        "mobile": recipient_data.mobile,
        "is_active": True
    })
    
    if existing:
        raise HTTPException(status_code=400, detail="Recipient with this mobile already exists")
    
    recipient = GreetingRecipient(
        tenant_id=user["tenant_id"],
        name=recipient_data.name,
        mobile=recipient_data.mobile,
        source_type=recipient_data.source_type,
        source_id=recipient_data.source_id,
        created_by=user["user_id"]
    )
    
    await db.greeting_recipients.insert_one(recipient.model_dump())
    
    return {"success": True, "recipient": recipient.model_dump()}


@router.post("/recipients/import-customers")
async def import_customers_as_recipients(request: Request):
    """
    Import existing customers as greeting recipients
    - Only imports customers with mobile numbers
    - Skips duplicates
    """
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    if user.get("role") not in ["super_admin", "tenant_admin"]:
        raise HTTPException(status_code=403, detail="Only Admin can import customers")
    
    # Get all customers for tenant
    customers = await db.customers.find(
        {"tenant_id": user["tenant_id"], "is_active": True},
        {"_id": 0, "id": 1, "name": 1, "phone": 1, "mobile": 1}
    ).to_list(1000)
    
    imported = 0
    skipped = 0
    
    for customer in customers:
        mobile = customer.get("mobile") or customer.get("phone")
        if not mobile:
            skipped += 1
            continue
        
        # Check if already exists
        existing = await db.greeting_recipients.find_one({
            "tenant_id": user["tenant_id"],
            "mobile": mobile,
            "is_active": True
        })
        
        if existing:
            skipped += 1
            continue
        
        recipient = GreetingRecipient(
            tenant_id=user["tenant_id"],
            name=customer.get("name", "Customer"),
            mobile=mobile,
            source_type="customer",
            source_id=customer["id"],
            created_by=user["user_id"]
        )
        
        await db.greeting_recipients.insert_one(recipient.model_dump())
        imported += 1
    
    return {
        "success": True,
        "imported": imported,
        "skipped": skipped,
        "message": f"Imported {imported} customers, skipped {skipped} (no mobile or duplicate)"
    }


@router.delete("/recipients/{recipient_id}")
async def remove_recipient(recipient_id: str, request: Request):
    """Remove a recipient from greeting list"""
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    result = await db.greeting_recipients.update_one(
        {"id": recipient_id, "tenant_id": user["tenant_id"]},
        {"$set": {"is_active": False}}
    )
    
    if result.modified_count == 0:
        raise HTTPException(status_code=404, detail="Recipient not found")
    
    return {"success": True, "message": "Recipient removed"}


@router.post("/recipients/{recipient_id}/opt-out")
async def opt_out_recipient(recipient_id: str, request: Request):
    """Mark recipient as opted out"""
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    result = await db.greeting_recipients.update_one(
        {"id": recipient_id, "tenant_id": user["tenant_id"]},
        {"$set": {"opted_out": True}}
    )
    
    if result.modified_count == 0:
        raise HTTPException(status_code=404, detail="Recipient not found")
    
    return {"success": True, "message": "Recipient opted out"}


# ==================== GREETING LOGS ====================

@router.get("/logs")
async def get_greeting_logs(
    request: Request,
    festival: Optional[str] = None,
    limit: int = 100
):
    """Get greeting send logs (audit only, NO analytics)"""
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    query = {"tenant_id": user["tenant_id"]}
    if festival:
        query["festival"] = festival
    
    logs = await db.greeting_logs.find(
        query, {"_id": 0}
    ).sort("sent_at", -1).limit(limit).to_list(limit)
    
    return {"success": True, "logs": logs}


# ==================== SEND GREETINGS (Internal/Cron) ====================

@router.post("/send")
async def send_greetings(
    request: Request,
    festival: str
):
    """
    Send greetings for a festival (called by cron or manual trigger)
    - ONLY republic_day or independence_day
    - Sends to all active, non-opted-out recipients
    """
    user = await get_current_user(request)
    db = get_db(request)
    
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
    
    if user.get("role") not in ["super_admin", "tenant_admin"]:
        raise HTTPException(status_code=403, detail="Only Admin can trigger greetings")
    
    if festival not in ["republic_day", "independence_day"]:
        raise HTTPException(
            status_code=400, 
            detail="Invalid festival. Only republic_day or independence_day allowed."
        )
    
    # Get config
    config = await db.festival_greeting_configs.find_one(
        {"tenant_id": user["tenant_id"]},
        {"_id": 0}
    )
    
    if not config or not config.get("is_enabled"):
        raise HTTPException(status_code=400, detail="Greetings are disabled for this tenant")
    
    # Get active recipients
    recipients = await db.greeting_recipients.find(
        {
            "tenant_id": user["tenant_id"],
            "is_active": True,
            "opted_out": False
        },
        {"_id": 0}
    ).to_list(10000)
    
    if not recipients:
        return {"success": True, "message": "No recipients to send", "sent": 0}
    
    # Prepare message
    message = GREETING_MESSAGES[festival].format(company_name=config["company_name"])
    today = datetime.now(timezone.utc).strftime("%Y-%m-%d")
    
    sent_count = 0
    failed_count = 0
    
    # Import SMS service
    try:
        from services.sms_service import send_sms
    except ImportError:
        send_sms = None
    
    for recipient in recipients:
        # Create log entry
        log = GreetingLog(
            tenant_id=user["tenant_id"],
            festival=festival,
            festival_date=today,
            recipient_id=recipient["id"],
            recipient_name=recipient["name"],
            recipient_mobile=recipient["mobile"],
            message=message
        )
        
        # Try to send SMS
        try:
            if send_sms:
                await send_sms(recipient["mobile"], message)
            log.status = "sent"
            sent_count += 1
        except Exception as e:
            log.status = "failed"
            log.error_message = str(e)
            failed_count += 1
        
        # Save log
        await db.greeting_logs.insert_one(log.model_dump())
    
    return {
        "success": True,
        "message": f"Greetings sent: {sent_count}, failed: {failed_count}",
        "sent": sent_count,
        "failed": failed_count,
        "total_recipients": len(recipients)
    }


# ==================== CRON TRIGGER CHECK ====================

@router.get("/check-today")
async def check_today_greetings():
    """
    Called by cron to check if today is a greeting day
    Returns list of tenants that need greetings sent
    """
    from fastapi import Request as FastAPIRequest
    
    today = datetime.now(timezone.utc).strftime("%m-%d")
    
    festival = None
    if today == "01-26":
        festival = "republic_day"
    elif today == "08-15":
        festival = "independence_day"
    
    if not festival:
        return {
            "is_greeting_day": False,
            "today": today,
            "next_greeting": "republic_day (01-26)" if today < "01-26" or today > "08-15" else "independence_day (08-15)"
        }
    
    return {
        "is_greeting_day": True,
        "festival": festival,
        "today": today,
        "message_template": GREETING_MESSAGES[festival]
    }
